libraries/Microsoft.Bot.Builder.Dialogs/DialogContext.cs

```diff
@@ -340,7 +341,7 @@ public async Task<DialogTurnResult> CancelAllDialogsAsync(CancellationToken canc
         /// `ResumeDialogAsync`, then the parent will end, too, and the result is passed to the next
         /// parent context.</remarks>
         /// <seealso cref="EndDialogAsync(object, CancellationToken)"/>
-        public async Task<DialogTurnResult> CancelAllDialogsAsync(string eventName, object eventValue = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<DialogTurnResult> CancelAllDialogsAsync(bool cancelParents, string eventName = null, object eventValue = null, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
>cancelParents [](start = 71, length = 13)

I think this is breaking change.  Is there a different way to do this?

Reply:
also, we need unit tests

---
In reply to: [335225667](https://github.com/microsoft/botbuilder-dotnet/pull/2744#discussion_r335225667) [](ancestors = 335225667)

Reply:
It's not a breaking change as its a completely new overload for 4.6. Prior to 4.6 this method took no arguments. I didn't add a unit tests because both signatures are already being tested.  The "false" case is the current 4.5 usage and the 4.6 usage with "true" is covered by the CancelAllDialogs step